dialer_test.go

```diff
@@ -0,0 +1,433 @@
+		salt[i] = 'a'
+	}
+	return salt
+}
+
+var (
+	opts = tarantool.Opts{
+		Timeout: 5 * time.Second,
+		//Concurrency: 32,
+		//RateLimit: 4*1024,
+	}
+
+	testDialUser    = "test"
+	testDialPass    = "test"
+	testDialVersion = [64]byte{'t', 'e', 's', 't'}
+
+	// Salt with end zeros.
+	testDialSalt = genSalt()
+
+	idRequestExpected = []byte{
+		0xce, 0x00, 0x00, 0x00, 29, // Length.
+		0x82, // Header map.
+		0x00, 0x49,
+		0x01, 0xce, 0x00, 0x00, 0x00, 0x00,
+
+		0x82, // Data map.
+		0x54,
+		0xcf, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x06, // Version.
+		0x55,
+		0x97, 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, // Features.
+	}
+
+	idResponseTyped = tarantool.ProtocolInfo{
+		Version:  6,
+		Features: []iproto.Feature{iproto.Feature(1), iproto.Feature(21)},
+		Auth:     tarantool.ChapSha1Auth,
+	}
+
+	idResponse = []byte{
+		0xce, 0x00, 0x00, 0x00, 37, // Length.
+		0x83, // Header map.
+		0x00, 0xce, 0x00, 0x00, 0x00, 0x00,
+		0x01, 0xce, 0x00, 0x00, 0x00, 0x00,
+		0x05, 0xce, 0x00, 0x00, 0x00, 0x61,
+
+		0x83, // Data map.
+		0x54,
+		0x06, // Version.
+		0x55,
+		0x92, 0x01, 0x15, // Features.
+		0x5b,
+		0xa9, 'c', 'h', 'a', 'p', '-', 's', 'h', 'a', '1',
+	}
+
+	idResponseNotSupported = []byte{
+		0xce, 0x00, 0x00, 0x00, 25, // Length.
+		0x83, // Header map.
+		0x00, 0xce, 0x00, 0x00, 0x80, 0x30,
+		0x01, 0xce, 0x00, 0x00, 0x00, 0x00,
+		0x05, 0xce, 0x00, 0x00, 0x00, 0x61,
+		0x81,
+		0x31,
+		0xa3, 'e', 'r', 'r',
+	}
+
+	authRequestExpectedChapSha1 = []byte{
+		0xce, 0x00, 0x00, 0x00, 57, // Length.
+		0x82, // Header map.
+		0x00, 0x07,
+		0x01, 0xce, 0x00, 0x00, 0x00, 0x00,
+
+		0x82, // Data map.
+		0xce, 0x00, 0x00, 0x00, 0x23,
+		0xa4, 't', 'e', 's', 't', // Login.
+		0xce, 0x00, 0x00, 0x00, 0x21,
+		0x92, // Tuple.
+		0xa9, 'c', 'h', 'a', 'p', '-', 's', 'h', 'a', '1',
+
+		// Scramble.
+		0xb4, 0x1b, 0xd4, 0x20, 0x45, 0x73, 0x22,
+		0xcf, 0xab, 0x05, 0x03, 0xf3, 0x89, 0x4b,
+		0xfe, 0xc7, 0x24, 0x5a, 0xe6, 0xe8, 0x31,
+	}
+
+	authRequestExpectedPapSha256 = []byte{
+		0xce, 0x00, 0x00, 0x00, 0x2a, // Length.
+		0x82, // Header map.
+		0x00, 0x07,
+		0x01, 0xce, 0x00, 0x00, 0x00, 0x00,
+
+		0x82, // Data map.
+		0xce, 0x00, 0x00, 0x00, 0x23,
+		0xa4, 't', 'e', 's', 't', // Login.
+		0xce, 0x00, 0x00, 0x00, 0x21,
+		0x92, // Tuple.
+		0xaa, 'p', 'a', 'p', '-', 's', 'h', 'a', '2', '5', '6',
+		0xa4, 't', 'e', 's', 't',
+	}
+
+	okResponse = []byte{
+		0xce, 0x00, 0x00, 0x00, 19, // Length.
+		0x83, // Header map.
+		0x00, 0xce, 0x00, 0x00, 0x00, 0x00,
+		0x01, 0xce, 0x00, 0x00, 0x00, 0x00,
+		0x05, 0xce, 0x00, 0x00, 0x00, 0x61,
+	}
+
+	errResponse = []byte{0xce}
+)
+
+type testDialOpts struct {
+	name                 string
+	wantErr              bool
+	expectedErr          string
+	expectedProtocolInfo tarantool.ProtocolInfo
+
+	// These options configure the behavior of the server.
+	isErrGreeting   bool
+	isErrId         bool
+	isIdUnsupported bool
+	isPapSha256Auth bool
+	isErrAuth       bool
+	isEmptyAuth     bool
```

Comment:
Do we use all options in the tests? You could remove some of them.

Reply:
Removed `isEmptyAuth` as the only one that wasn't used in the tests.